moveit_core/robot_state/src/cartesian_interpolator.cpp

```diff
@@ -42,20 +42,126 @@
+                                             const moveit::core::JointModelGroup* group, double jump_threshold_factor)
+{
+  if (waypoints.size() < MIN_STEPS_FOR_JUMP_THRESH)
+  {
+    RCLCPP_WARN(LOGGER,
+                "The computed path is too short to detect jumps in joint-space. "
+                "Need at least %zu steps, only got %zu. Try a lower max_step.",
+                MIN_STEPS_FOR_JUMP_THRESH, waypoints.size());
+  }
+
+  std::vector<double> dist_vector;
+  dist_vector.reserve(waypoints.size() - 1);
+  double total_dist = 0.0;
+  for (std::size_t i = 1; i < waypoints.size(); ++i)
+  {
+    double dist_prev_point = waypoints[i]->distance(*waypoints[i - 1], group);
+    dist_vector.push_back(dist_prev_point);
+    total_dist += dist_prev_point;
+  }
+
+  // compute the average distance between the states we looked at.
+  double thres = jump_threshold_factor * (total_dist / static_cast<double>(dist_vector.size()));
+  for (std::size_t i = 0; i < dist_vector.size(); ++i)
+  {
+    if (dist_vector[i] > thres)
+    {
+      return i + 1;
+    }
+  }
+
+  return std::nullopt;
+}
+
+std::optional<int> hasAbsoluteJointSpaceJump(const std::vector<moveit::core::RobotStatePtr>& waypoints,
+                                             const moveit::core::JointModelGroup* group, double revolute_threshold,
+                                             double prismatic_threshold)
+{
+  const bool check_revolute = revolute_threshold > 0.0;
+  const bool check_prismatic = prismatic_threshold > 0.0;
+
+  const std::vector<const moveit::core::JointModel*>& joints = group->getActiveJointModels();
+  for (std::size_t i = 1; i < waypoints.size(); ++i)
+  {
+    for (const auto& joint : joints)
+    {
+      const double distance = waypoints[i]->distance(*waypoints[i - 1], joint);
+      switch (joint->getType())
+      {
+        case moveit::core::JointModel::REVOLUTE:
+          if (check_revolute && distance > revolute_threshold)
+          {
+            return i;
+          }
+          break;
+        case moveit::core::JointModel::PRISMATIC:
+          if (check_prismatic && distance > prismatic_threshold)
+          {
+            return i;
+          }
+          break;
+        default:
+          RCLCPP_WARN(LOGGER,
+                      "Joint %s has not supported type %s. \n"
+                      "hasAbsoluteJointSpaceJump only supports prismatic and revolute joints.",
+                      joint->getName().c_str(), joint->getTypeName().c_str());
+          continue;
+      }
+    }
+  }
+
+  return std::nullopt;
+}
+}  // namespace
+
+JumpThreshold JumpThreshold::disabled()
+{
+  return JumpThreshold();
+}
+
+JumpThreshold JumpThreshold::relative(double relative_factor)
+{
+  JumpThreshold threshold;
+  threshold.relative_factor = relative_factor;
+  return threshold;
+}
+
+JumpThreshold JumpThreshold::absolute(double revolute, double prismatic)
+{
+  JumpThreshold threshold;
+  threshold.revolute = revolute;
```

Comment:
Same here: Both values should be positive right?

Reply:
Added a check.